### ISSUE TITLE
Toggle keyboard support?

### ISSUE THREAD

**Issue by SuneRadich** [openseadragon/openseadragon#794]
I have trawled through the documentation, looking for a way to temporarily turn off keyboard support, but have not found any toggles I could use.

Basically what I want is to let the user select part of the image (using openseadragonselection plugin) and then let the user adjust the selected area using the arrow keys. But when I do that, the image also moves in the viewer.

Is there a way to toggle this behaviour - or to redefine the keys?

**Comment by iangilman** [comment]
Looks like we don't have an official option for that; would probably be good to add. Would you be up for writing a patch for that? 

Meanwhile, here is a hack: 

```
viewer.innerTracker.keyDownHandler = null;
viewer.innerTracker.keyPressHandler = null;
```

**Comment by shubhekshajalan** [comment]
Hey, I'd like to work on this. As I am a beginner and new to the codebase, could you point me to the file(s) that need to be changed?

**Comment by iangilman** [comment]
@shubhekshajalan Wonderful! This is where the key handling happens: 

https://github.com/openseadragon/openseadragon/blob/master/src/viewer.js#L2285-L2370

You'll want to define a new option to use. You can add the documentation here: 

https://github.com/openseadragon/openseadragon/blob/master/src/openseadragon.js#L365

...and the default value here: 

https://github.com/openseadragon/openseadragon/blob/master/src/openseadragon.js#L1035

Let me know if you have questions!

**Comment by cabooserwar** [comment]
Hey! I'm a newbie, and I have a feeling the ball may have been dropped on this, so I may take a look.

**Comment by iangilman** [comment]
@cabooserwar Yes, looks like the ball has been dropped... thank you for taking a look!

**Comment by jetic83** [comment]
Hi, I have the same feature request. I put overlays on the OSD which contain input elements, such the user should be able to type text on the overlays. Whenever a a, s, d, or w is typed, the image moves, which should be switched off.
Any progress here?

**Comment by jetic83** [comment]
In the meanwhile, this worked for me

`viewer.innerTracker.keyHandler = null;`

**Comment by iangilman** [comment]
@jetic83 It appears the ball has been dropped once again... you're welcome to dive in and make a patch!

**Comment by wmavis** [comment]
It looks like there is a PR for this one?

**Comment by iangilman** [comment]
This PR? https://github.com/colinrlim/openseadragon/pull/1

Looks great! I didn't see it, because it's not filed against this repo. I'll ask @colinrlim to switch it over here. 

@wmavis Thank you for catching it!

**Comment by achu1998** [comment]
Is this issue still open?

**Comment by Harshit9026** [comment]
Hi! I’m interested in working on this issue.
Could you please assign it to me?
I’ll make sure to follow the project’s contribution guidelines.
Thanks!

**Comment by iangilman** [comment]
@Harshit9026 Great, welcome! I've assigned it to you. You might take a look at https://github.com/colinrlim/openseadragon/pull/1.

**Comment by akriti04gupta** [comment]
Hi ! 👋

I'm interested in working on this issue and would love to contribute a patch to add support for toggling keyboard navigation using a configuration option.

Could you please assign this issue to me?

Thanks in advance! 😊

**Comment by iangilman** [comment]
@akriti04gupta Sure... looks like @Harshit9026 has walked away from it.

**Comment by iangilman** [comment]
Fixed by #2773
